### PR TITLE
fix: do not trigger TZ resolution for VTIMEZONE observance DTSTART

### LINKED ISSUE #495: node-ical 0.25.6: Crash parsing VTIMEZONE with DTSTART year 0001 (RangeError "Cannot parse: 1-01-15T12:00:00Z")
# Parsing VTIMEZONE with DTSTART year 0001 fails in 0.25.6

I found a parsing regression in `node-ical` that affects ICS feeds containing a `VTIMEZONE` definition with `STANDARD` / `DAYLIGHT` `DTSTART` values in year `0001`.

This worked in my MagicMirror setup with `node-ical@0.22.1` and fails in `node-ical@0.25.6`.

## Summary

`node-ical@0.25.6` throws:

```text
Cannot parse: 1-01-15T12:00:00Z
```

The failure happens while resolving a `VTIMEZONE` block, before the calendar events can be processed.

## Test Environment

- `node-ical@0.22.1`: parses successfully
- `node-ical@0.25.5`: parses successfully
- `node-ical@0.25.6`: fails
- Node.js: `v24.14.1`

The issue was discovered through MagicMirror's default `calendar` module after upgrading from MagicMirror `2.34.0` to `2.35.0`, which updates `node-ical` from `0.22.1` to `0.25.6`.

## Actual error

Stack trace:

```text
RangeError: Cannot parse: 1-01-15T12:00:00Z
    at We (file:///opt/magic_mirror/node_modules/temporal-polyfill/chunks/internal.js:1562:11)
    at toInstantSlots (file:///opt/magic_mirror/node_modules/temporal-polyfill/chunks/classApi.js:249:10)
    at Instant.from (file:///opt/magic_mirror/node_modules/temporal-polyfill/chunks/classApi.js:563:17)
    at Object.resolveVTimezoneToIana (/opt/magic_mirror/node_modules/node-ical/lib/tz-utils.js:582:37)
    at fallbackWithStackTimezone (/opt/magic_mirror/node_modules/node-ical/ical.js:403:37)
    at /opt/magic_mirror/node_modules/node-ical/ical.js:490:23
    at Object.DTSTART (/opt/magic_mirror/node_modules/node-ical/ical.js:1005:38)
    at Object.handleObject (/opt/magic_mirror/node_modules/node-ical/ical.js:1051:39)
    at Object.parseLines (/opt/magic_mirror/node_modules/node-ical/ical.js:1121:20)
    at Object.parseICS (/opt/magic_mirror/node_modules/node-ical/ical.js:1187:19)
```

## Relevant ICS fragment

The failing calendars contain this `VTIMEZONE` block:

```ics
BEGIN:VTIMEZONE
TZID:W. Europe Standard Time
X-EM-DISPLAYNAME:(UTC+01:00) Amsterdam\, Berlin\, Bern\, Rom\, Stockholm\, Wien
BEGIN:DAYLIGHT
TZNAME:Mitteleuropäische Sommerzeit
DTSTART:00010325T020000
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3
END:DAYLIGHT
BEGIN:STANDARD
TZNAME:Mitteleuropäische Zeit
DTSTART:00011028T030000
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
END:STANDARD
END:VTIMEZONE
```

## Notes

- The same ICS parses successfully with `node-ical@0.22.1`.
- The same ICS fails with `node-ical@0.25.6`.
- If I remove only the `VTIMEZONE` block above from the failing ICS file, `node-ical@0.25.6` parses the rest of the file successfully.
- The failure happens while resolving timezone metadata, before calendar events can be processed.
- In my tests, the smallest reproducing input requires two `VTIMEZONE` blocks: one for `Europe/Berlin` and one for `W. Europe Standard Time`.
- In that minimal reproduction, no `VEVENT` is required, and no event needs to reference `W. Europe Standard Time`.
- This regression may be related to the `VTIMEZONE` resolution logic introduced for [#478](https://github.com/jens-maus/node-ical/issues/478) and released in `0.25.6`.
- I tested `node-ical@0.25.5` against the same ICS: it parses successfully.
- The issue reproduces on `node-ical@0.25.6`: `Cannot parse: 1-01-15T12:00:00Z`.

## RFC note

According to RFC 5545:

- `date-fullyear = 4DIGIT`
- `DATE-TIME = date "T" time`
- `STANDARD` and `DAYLIGHT` in `VTIMEZONE` must include `DTSTART`

So `DTSTART:00010325T020000` appears to be syntactically valid as a 4-digit year. It is unusual, but it does not seem invalid per RFC 5545.

## Minimal reproduction

```js
const ical = require("node-ical");

const ics = `BEGIN:VCALENDAR
VERSION:2.0
BEGIN:VTIMEZONE
TZID:Europe/Berlin
BEGIN:DAYLIGHT
DTSTART:19810329T020000
RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU
TZNAME:MESZ
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
END:DAYLIGHT
BEGIN:STANDARD
DTSTART:19961027T030000
RRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU
TZNAME:MEZ
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
END:STANDARD
END:VTIMEZONE
BEGIN:VTIMEZONE
TZID:W. Europe Standard Time
X-EM-DISPLAYNAME:(UTC+01:00) Amsterdam\, Berlin\, Bern\, Rom\, Stockholm\, Wien
BEGIN:DAYLIGHT
TZNAME:Mitteleuropäische Sommerzeit
DTSTART:00010325T020000
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3
END:DAYLIGHT
BEGIN:STANDARD
TZNAME:Mitteleuropäische Zeit
DTSTART:00011028T030000
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
END:STANDARD
END:VTIMEZONE
END:VCALENDAR`;

console.log(ical.parseICS(ics));
```

Tested result:

- `node-ical@0.25.5`: parses successfully
- `node-ical@0.25.6`: fails with `Cannot parse: 1-01-15T12:00:00Z`

This tested minimal reproduction requires both `VTIMEZONE` blocks shown above. In my tests, the `W. Europe Standard Time` block alone did not reproduce the issue.

## Expected behavior

The ICS should parse successfully, or at minimum the timezone resolution should fail gracefully without crashing the entire parse.

### PR DESCRIPTION
**Pull request by KristjanESPERANTO**
### Problem

ICS files originating from Outlook or Exchange often contain Microsoft-style VTIMEZONE blocks with year 0001 in DTSTART. These can also appear in other calendar systems (Nextcloud, Google Calendar, etc.) when events are imported or shared from Microsoft sources.

```
BEGIN:STANDARD
DTSTART:00010325T020000
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
END:STANDARD
```

Year 0001 is Microsoft's convention for "this rule has always been in effect". This is valid per RFC 5545.

Since node-ical 0.25.x, parsing these files throws:
```
RangeError: Cannot parse: 1-01-15T12:00:00Z
```

The generic DTSTART handler was treating observance DTSTARTs the same as event DTSTARTs — trying to resolve the enclosing VTIMEZONE to an IANA zone. That resolution builds a probe timestamp from the year (`parseInt('0001') → 1`), producing the malformed ISO string `1-01-15T12:00:00Z` which `Temporal.Instant.from()` rejects.

### Fix

**Root cause** (`ical.js`): DTSTART inside STANDARD/DAYLIGHT sub-components is now recognized as a local wall-clock time for the observance rule. It skips timezone resolution entirely, which is the correct behavior — these timestamps define *when* a DST rule takes effect, not an event in a timezone.

**Defensive guard** (`lib/tz-utils.js`): If `resolveVTimezoneToIana` is ever called with a year < 1000, the probe year is clamped to 1970 and zero-padded to 4 digits. IANA zone data before 1970 is unreliable anyway (RFC 5545 itself only guarantees data since 1970), so this has no practical impact on zone matching.

### Tests

- Unit test: `resolveVTimezoneToIana` with `year=1` no longer throws
- Integration test: full ICS file with year-0001 VTIMEZONE parses correctly and produces the expected UTC timestamp
- ICS fixture: two VTIMEZONE blocks (one normal, one with year 0001) + a VEVENT referencing the problematic zone

Fixes #495

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented crashes and parsing errors for iCalendar timezone definitions with very old years (e.g., year 0001).
  * Improved handling of DTSTART values inside timezone observance blocks so event times are resolved correctly.

* **Tests**
  * Added regression and unit tests covering VTIMEZONE/DTSTART edge cases with extremely old years.

### CODE REVIEW

File: ical.js [hunk 1]
```diff
@@ -376,6 +376,12 @@ const dateParameter = function (name) {
         // GMT
         newDate = new Date(Date.UTC(year, monthIndex, day, hour, minute, second));
         tzUtil.attachTz(newDate, 'Etc/UTC');
+      } else if (curr.type === 'STANDARD' || curr.type === 'DAYLIGHT') {
+        // Inside a VTIMEZONE observance block the DTSTART is a plain local
+        // wall-clock time that defines when the rule takes effect — it must
+        // NOT trigger timezone resolution (which would look up the *enclosing*
+        // VTIMEZONE and could crash on exotic years like 0001).
+        newDate = new Date(year, monthIndex, day, hour, minute, second);
```
[the remark this answers is not in the document]
Comment (KristjanESPERANTO) [reply ical.js:384]: Done.

**Comment by KristjanESPERANTO** [comment]
@jens-maus Coderabbit seems satisfied. If you are fine with this changes, a new release would be nice after merging 🙂
